src/client/components/MealList.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from "react";
+import RingLoader from "react-spinners/RingLoader"; //npm install --save react-spinners
+import MealsCSS from "./meals.module.css";
+
+const MealList = () => {
+  const [listOfMeals, setListOfMeals] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  // Loading spinner
+
+  useEffect(() => {
+    setLoading(true);
+    setTimeout(() => {
+      setLoading(false);
+    }, 1000);
+  }, []);
+
+  useEffect(() => {
+    (async () => {
+      const data = await fetch("api/meals").then((data) => data.json());
+      setListOfMeals(data);
+    })();
```

Comment:
Instead of using a `setTimeOut` in `useEffect` to display the RingLoader, You can use it the Fetch useEffect by making it synchronized. And that is the purpose we need a Loader, to show that meals are loading.